pxtlib/discourse.ts

```diff
@@ -0,0 +1,72 @@
+namespace pxt.discourse {
+    interface DiscoursePostResponse {
+        featured_link?: string;
+        post_stream?: DiscoursePostStream;
+    }
+    interface DiscoursePostStream {
+        posts?: DiscoursePost[];
+    }
+    interface DiscoursePost {
+        link_counts?: DiscourseLinkCount[];
+    }
+    interface DiscourseLinkCount {
+        url?: string;
+    }
+    interface TagTopic {
+        id: string;
+        title: string;
+        image_url: string;
+        slug: string;
+        views: number;
+        like_count: number;
+        posters: {
+            user_id: string;
+        }[];
+    }
+    interface TagUser {
+        id: number;
+        username: string;
+        name: string;
+        avatar_template: string;
+    }
+    interface TagsResponse {
+        users: TagUser[];
+        topic_list: {
+            topics: TagTopic[];
+        }
+    }
+
+    export function extractSharedIdFromPostUrl(url: string): Promise<string> {
+        // https://docs.discourse.org/#tag/Posts%2Fpaths%2F~1posts~1%7Bid%7D.json%2Fget
+        return pxt.Util.httpGetJsonAsync(url + ".json")
+            .then((json: DiscoursePostResponse) => {
+                // extract from post_stream
+                let projectId = json.post_stream
+                    && json.post_stream.posts
```

Comment:
we probably ought to update tslint (/ move to eslint if easy enough, as tslint is deprecated?), would be great to be able to use optional chaining in places like this / the million `pxt.appTarget.appTheme && ...`s (right now it marks that as an error as it uses an older typescript)

Reply:
not in this pr